Discussion/Discussion/Presentation/CreateNewThread/CreateNewThreadViewModel.swift

```diff
@@ -81,4 +87,19 @@ public class CreateNewThreadViewModel: ObservableObject {
             return false
         }
     }
+    
+    func trackCreateNewThread(
+        courseID: String,
+        topicID: String,
+        postType: String,
+        followPost: Bool
+    ) {
+        analytics?.discussionCreateNewPost(
+            courseID: courseID,
+            topicID: topicID,
+            postType: postType,
+            followPost: followPost,
+            author: storage?.user?.username ?? ""
```

Comment:
In `trackCommentAdded(` & `trackResponseAdded(`, we are using `author` as `newComment.authorName` but in `analytics?.discussionCreateNewPost(`, we are using `storage?.user?.username` as an `author` attribute. Is it intentional?

Reply:
Yes it's intentional.

> In trackCommentAdded( & trackResponseAdded(, we are using author as newComment.authorName
At these locations, new comment and new response object is available so authorname is available there so using 

> but in analytics?.discussionCreateNewPost(, we are using storage?.user?.username as an author attribute. Is it intentional?

Newly created thread object is not available so using the logged in user detail as user created the thread.